src/queue.js

```diff
@@ -0,0 +1,98 @@
+let instance = null
+
+class Queue {
+
+  constructor() {
+    if(!instance){
+      instance = this // eslint-disable-line consistent-this
+      this.setup()
+    }
+    return instance
+  }
+
+  setup() {
+    this.tasks = []
+    this.pendingCounter = {}
+    this.ranCounter = {}
+    this.running = false
+  }
+
+  valid(task) {
+    if(
+      typeof task.run !== 'function' ||
+      typeof task.error !== 'function' ||
+      typeof task.done !== 'function'
+    ) throw new Error('Task has an invalid object signature, methods required')
```

Comment:
Actually, `Queue.valid` should not throw, but return `false` instead. It's clearly a checking method, so the a developer using the class would expect it to return false for an object without a proper interface.